### ISSUE TITLE
Please replace wget with curl as the default downloader for Windows

### ISSUE THREAD

**Issue by hishamhm** [keplerproject/luarocks#273]
@leafo added HTTPS support for MoonRocks. I've just updated the code of `luarocks upload` for LuaRocks 2.2.0 to work with HTTPS, but for that we either need LuaSec (unavailable on Windows because of `cfg.fs_use_modules=false`) or Curl. Unfortunately Wget does not support `multipart/form-data` (see wget manpage), so we need a new default on Windows.

I see a bunch of binaries available here ( http://curl.haxx.se/download.html ) but I don't know which one to pick, and can't test which of the necessary libraries work (see at the bottom of that link).

I will commit the code for Curl support on `luarocks upload` now. To test it, you can create a dummy rockspec and upload it and then delete it on http://rocks.moonscript.org (just take an existing rockspec, rename it and edit the `package` field accordingly).

Thank you!

**Comment by ignacio** [comment]
Ok, sorry. Took a while. It is not working reliably, but for different reasons.
Now, luarocks upload --force is working. But not every time. Sometimes, it fails in the following way:

~~~~
D:\Temp\try\try-0.1.0-1\lua-try-0.1.0\rockspecs>luarocks upload try-0.1.0-1.rockspec --force
C:/Users/Ignacio/AppData/Roaming/luarocks/upload_config.lua
Enabling api.debug
Sending try-0.1.0-1.rockspec ...
[GET via curl] https://rocks.moonscript.org/api/tool_version?current=1%2e0%2e0 ...

Error: API failure: nil - https://rocks.moonscript.org/api/1/{snipped my api key}/check_rockspec?package=try&version=0%2e1%2e0%2d1
~~~~

On a side note, wouldn't it be better to not print the api key (indirectly)? I almost posted mine without realizing when submiting the above.

**Comment by ignacio** [comment]
Running curl from the command line, I usually get it work many times in a row and then it fails, many times in a row. It seems to be something with the server.
Maybe @leafo can shed some light on this?

~~~~
"curl"  -k -v -L --user-agent "LuaRocks/2.2.1 win32-x86 via curl" --connect-timeout 30 "https://rocks.moonscript.org/api/tool_version?current=1"%"2e0"%"2e0"
* About to connect() to rocks.moonscript.org port 443 (#0)
*   Trying 54.204.57.69...
* connected
* Connected to rocks.moonscript.org (54.204.57.69) port 443 (#0)
* SSLv3, TLS handshake, Client hello (1):
* SSL connection timeout
* Closing connection #0
curl: (28) SSL connection timeout
~~~~

**Comment by leafo** [comment]
I ran it a few dozen times but was unable to get it to time out. Are you still having the issue?

**Comment by ignacio** [comment]
Yes. Intermitently but happens a lot during the da y. Will try again soon and report back.

**Comment by ignacio** [comment]
Just tried a lot of times from the computer at home. I can't reproduce the issue. Will try again tomorrow when I'm at work. It seems that it might be something on my end.

**Comment by ignacio** [comment]
Well, it seems that the issue has to do with crap network here at work.
I've been using LR 2.2.1 with curl and it is working fine. 

So, things left to do, decide which version of curl to bundle.

**Comment by hishamhm** [comment]
Cool, thanks for testing!

**Comment by justinmk** [comment]
In Neovim (neovim/neovim@2fbc42aa8a0a) we bundle `curl.exe` and `ca-bundle.crt`. I found   https://winampplugins.co.uk/curl/ to be a good source for those artifacts.
